File: dotnet/Version1.cs
using System;
using System.ServiceModel;

// Version 1: Consume SOAP service directly
// Usage: dotnet run --project Version1.csproj -- <number>
// Example: dotnet run --project Version1.csproj -- 10

namespace NumberConversion
{
    [ServiceContract]
    public interface INumberConversion
    {
        [OperationContract]
        string NumberToWords(ulong ubiNum);
    }

    class Program
    {
        static void Main(string[] args)
        {
            string number = args.Length > 0 ? args[0] : "10";

            BasicHttpBinding binding = new BasicHttpBinding();
            EndpointAddress address = new EndpointAddress("https://www.dataaccess.com/webservicesserver/NumberConversion.wso");
            
            ChannelFactory<INumberConversion> factory = new ChannelFactory<INumberConversion>(binding, address);
            INumberConversion client = factory.CreateChannel();

            try
            {
                string result = client.NumberToWords(ulong.Parse(number));
                Console.WriteLine(result);
            }
            catch (Exception ex)
            {
                Console.WriteLine($"Error: {ex.Message}");
            }
        }
    }
}

File: java/Version1.java
import javax.xml.namespace.QName;
import jakarta.xml.ws.Service;
import java.net.URL;

// Version 1: Consume SOAP service directly
// Usage: javac Version1.java && java Version1 <number>
// Example: java Version1 10

public class Version1 {
    public static void main(String[] args) {
        String number = args.length > 0 ? args[0] : "10";
        
        try {
            URL url = new URL("https://www.dataaccess.com/webservicesserver/NumberConversion.wso?WSDL");
            QName qname = new QName("http://www.dataaccess.com/webservicesserver/", "NumberConversion");
            
            Service service = Service.create(url, qname);
            NumberConversion proxy = service.getPort(NumberConversion.class);
            
            String result = proxy.numberToWords(Long.parseLong(number));
            System.out.println(result);
        } catch (Exception e) {
            System.err.println("Error: " + e.getMessage());
        }
    }
}

interface NumberConversion {
    String numberToWords(long ubiNum);
}
